platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/extensions/property/PropertyUpgradeClassSourceGenerator.java

```diff
@@ -78,26 +85,67 @@ private static MethodSpec mapToMethodSpec(CallInterceptionRequest request, Consu
             .orElseThrow(() -> new RuntimeException(PropertyUpgradeRequestExtra.class.getSimpleName() + " should be present at this stage!"));
 
         try {
-            CallableInfo callable = request.getInterceptedCallable();
             ImplementationInfo implementation = request.getImplementationInfo();
-            List<ParameterSpec> parameters = callable.getParameters().stream()
-                .map(parameter -> ParameterSpec.builder(typeName(parameter.getParameterType()), parameter.getName()).build())
-                .collect(Collectors.toList());
-            MethodSpec spec = MethodSpec.methodBuilder(implementation.getName())
-                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
-                .addParameter(typeName(callable.getOwner().getType()), SELF_PARAMETER_NAME)
-                .addParameters(parameters)
-                .addCode(generateMethodBody(implementation, callable, implementationExtra))
-                .returns(typeName(callable.getReturnType().getType()))
-                .addAnnotations(getAnnotations(implementationExtra))
-                .build();
+            CallableInfo callable = request.getInterceptedCallable();
+
+            MethodSpec spec;
+            if (implementationExtra.getBridgedMethod() != null) {
+                spec = mapToBridgedMethod(implementation.getName(), implementationExtra, callable);
+            } else {
+                List<ParameterSpec> parameters = callable.getParameters().stream()
+                    .map(parameter -> ParameterSpec.builder(typeName(parameter.getParameterType()), parameter.getName()).build())
+                    .collect(Collectors.toList());
+                spec = MethodSpec.methodBuilder(implementation.getName())
+                    .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                    .addParameter(typeName(callable.getOwner().getType()), SELF_PARAMETER_NAME)
+                    .addParameters(parameters)
+                    .addCode(generateMethodBody(implementation, callable, implementationExtra))
+                    .returns(typeName(callable.getReturnType().getType()))
+                    .addAnnotations(getAnnotations(implementationExtra))
+                    .build();
+            }
+
             onProcessedRequest.accept(request);
             return spec;
         } catch (Exception e) {
             onFailure.accept(new HasFailures.FailureInfo(request, e.getMessage()));
             throw e;
         }
+    }
+
+    private static MethodSpec mapToBridgedMethod(String methodName, PropertyUpgradeRequestExtra implementationExtra, CallableInfo callable) {
```

Comment:
💭 Here is a part where we generate `$$BridgeFor$$` methods that are public and delegate to Adapter methods